spec/requests/demo_mode/sessions_spec.rb

```diff
@@ -72,6 +75,36 @@
           expect(response_json['password']).to be_nil
         end
       end
+
+      context 'with option' do
+        before do
+          DemoMode.configure do
+            around_persona_generation do |_, options|
+              if options.present? && options[:example_custom_option].present?
+                DummyUser.create!(name: options[:example_custom_option][:name])
+              end
+            end
```

Comment:
The current implementation appears to effectively bypass the persona entirely, so it's not particularly realistic. Thoughts on utilizing the first "generator" argument to invoke the persona?

```suggestion
            around_persona_generation do |generator, options|
              generator.call.tap do |dummy_user|
                if options.present? && options[:example_custom_option].present?
                  dummy_user.update!(name: options[:example_custom_option][:name])
                end
              end
            end
```

Note: In more complex vendoring apps, personas may produce different _types_ (e.g. user personas and [non-user personas](https://github.com/Betterment/demo_mode?tab=readme-ov-file#non-user-personas)). That doesn't seem necessary to incorporate into your test here, but that may be something to explore in those apps, like whether it requires more conditional logic in this block, is a signal it should be pushed down into reusable logic in the personas, etc.